### ISSUE TITLE
Functions CRDs and operator

### ISSUE THREAD

**Issue by yairyairyair** [knative/func#1680]
Hi,

just like we have Service kind at serving.knative.dev/v1, i would like either a Function kind or to add a function building to image capability to the existing Service kind.

example idea (can use either codeURI or code):

apiVersion: serving.knative.dev/v1
kind: Function
metadata:
  name: helloworld-go
  namespace: default
spec:
  template:
    spec:
      codeURI: https://github.com/some-example/example-serverless-project
      code: |
       module.exports = (req,res)=>res.send('hello world')
      env:
        - name: TARGET
          value: "Go Sample v1"

Bottom line,That way instead of using kn func deploy we could just kubectl apply a function and the thats it.
The tekton pipeline will build it to an image and knative serving will serve it.
